cpp/src/tsne/distances.cuh

```diff
@@ -41,15 +41,10 @@ void get_distances(const float *X, const int n, const int p, long *indices,
                    cudaStream_t stream) {
   // TODO: for TSNE transform first fit some points then transform with 1/(1+d^2)
   // #861
-  float **knn_input = new float *[1];
-  int *sizes = new int[1];
-  knn_input[0] = (float *)X;
-  sizes[0] = n;
 
-  std::vector<float *> input_vec(1);
-  std::vector<int> sizes_vec(1);
-  input_vec.push_back(knn_input[0]);
-  sizes_vec.push_back(sizes[0]);
+  // TODO make brute_force_knn take a const float*
```

Comment:
We could add a convenience wrapper for the brute force kNN that accepts a single float* and calls the version that accepts multiple. Would you like to do that in this PR?

Reply:
Oops, unclear comment. I was only trying to avoid the `const_cast`s here and on line 58R, not the `vector`. But it looks like making it const-correct would be hard, so I've removed the comment.